pkg/acquisition/modules/syslog/internal/parser/rfc5424/parse.go

```diff
@@ -0,0 +1,398 @@
+	return nil
+}
+
+func (r *RFC5424) parseVersion() error {
+	if r.buf[r.position] != '1' {
+		return fmt.Errorf("version must be 1")
+	}
+	r.position += 2
+	if r.position >= r.len {
+		return fmt.Errorf("version must be followed by a space")
+	}
+	return nil
+}
+
+func (r *RFC5424) parseTimestamp() error {
+
+	timestamp := []byte{}
+
+	if r.buf[r.position] == NIL_VALUE {
+		r.Timestamp = time.Now().UTC().Round(0)
+		r.position += 2
+		return nil
+	}
+
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == ' ' {
+			break
+		}
+		timestamp = append(timestamp, c)
+		r.position++
+	}
+
+	if len(timestamp) == 0 {
+		return fmt.Errorf("timestamp is empty")
+	}
+
+	if r.position == r.len {
+		return fmt.Errorf("EOL after timestamp")
+	}
+
+	date, err := time.Parse(VALID_TIMESTAMP, string(timestamp))
+
+	if err != nil {
+		return fmt.Errorf("timestamp is not valid")
+	}
+
+	r.Timestamp = date
+
+	r.position++
+
+	if r.position >= r.len {
+		return fmt.Errorf("EOL after timestamp")
+	}
+
+	return nil
+}
+
+func (r *RFC5424) parseHostname() error {
+	if r.buf[r.position] == NIL_VALUE {
+		r.Hostname = ""
+		r.position += 2
+		return nil
+	}
+
+	hostname := []byte{}
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == ' ' {
+			r.position++
+			break
+		}
+		hostname = append(hostname, c)
+		r.position++
+	}
+	if r.strictHostname {
+		if !utils.IsValidHostnameOrIP(string(hostname)) {
+			return fmt.Errorf("hostname is not valid")
+		}
+	}
+	if len(hostname) == 0 {
+		return fmt.Errorf("hostname is empty")
+	}
+	r.Hostname = string(hostname)
+	return nil
+}
+
+func (r *RFC5424) parseAppName() error {
+	if r.buf[r.position] == NIL_VALUE {
+		r.Tag = ""
+		r.position += 2
+		return nil
+	}
+
+	appname := []byte{}
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == ' ' {
+			r.position++
+			break
+		}
+		appname = append(appname, c)
+		r.position++
+	}
+
+	if len(appname) == 0 {
+		return fmt.Errorf("appname is empty")
+	}
+
+	if len(appname) > 48 {
+		return fmt.Errorf("appname is too long")
+	}
+
+	r.Tag = string(appname)
+	return nil
+}
+
+func (r *RFC5424) parseProcID() error {
+	if r.buf[r.position] == NIL_VALUE {
+		r.PID = ""
+		r.position += 2
+		return nil
+	}
+
+	procid := []byte{}
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == ' ' {
+			r.position++
+			break
+		}
+		procid = append(procid, c)
+		r.position++
+	}
+
+	if len(procid) == 0 {
+		return fmt.Errorf("procid is empty")
+	}
+
+	if len(procid) > 128 {
+		return fmt.Errorf("procid is too long")
+	}
+
+	r.PID = string(procid)
+	return nil
+}
+
+func (r *RFC5424) parseMsgID() error {
+	if r.buf[r.position] == NIL_VALUE {
+		r.MsgID = ""
+		r.position += 2
+		return nil
+	}
+
+	msgid := []byte{}
+	for r.position < r.len {
+		c := r.buf[r.position]
+		if c == ' ' {
+			r.position++
+			break
+		}
+		msgid = append(msgid, c)
+		r.position++
+	}
+
+	if len(msgid) == 0 {
+		return fmt.Errorf("msgid is empty")
+	}
+
+	if len(msgid) > 128 {
```

Comment:
from https://www.rfc-editor.org/rfc/rfc5424.html#section-6.2.3 seems MSGID can be only up to 32 CHARS